cluster-api/Makefile

```diff
@@ -12,14 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY: genapi genconversion
+.PHONY: genapi clean
 
-all: genapi genconversion
+APIBUILDER_GENERATORS=apiregister conversion openapi
```

Comment:
Can we actually just break out all the generator to separate rules? Some thing like (untested):

```Makefile

APIBUILDER_IMAGE_TAG=dev-cluster-api-generator
GENERATOR=

.PHONY: .buildimage
.buildimage:
    docker build --build-arg DEVUSERNAME=`id -un` --build-arg DEVUSERID=`id -u` -f Dockerfile.apigen . -t ${APIBUILDER_IMAGE_TAG}

.PHONY: .gencode
.gencode: .buildimage
    docker run --rm --user=`id -un` -v ${PWD}:/go/src/k8s.io/kube-deploy/cluster-api ${APIBUILDER_IMAGE_TAG} /opt/apiserver-builder/bin/apiserver-boot build generated --generator ${GENERATOR}

.PHONY: gen-apiregister
gen-apiregister: GENERATOR=apiregister
gen-apiregister: .gencode

.PHONY: gen-conversion
gen-conversion: GENERATOR=conversion
gen-conversion: .gencode

.PHONY: gen-openapi
gen-openapi: GENERATOR=openapi
gen-openapi: .gencode

.PHONY: gen-deepcopy
gen-deepcopy: GENERATOR=deepcopy
gen-deepcopy: .gencode

.PHONY: gen-client
gen-client: GENERATOR=client
gen-client: .gencode

.PHONY: gen-all
gen-all: gen-deepcopy gen-conversion gen-openapi gen-client gen-apiregister
```

I may have missed some generators. The reason is some generators are needed way more often and other currently generate conflicts (apiregister).